src/bootloaders/shim-systemd.c

```diff
@@ -0,0 +1,254 @@
+
+#include <linux/limits.h>
+
+#include "bootloader.h"
+#include "bootman.h"
+#include "bootvar.h"
+#include "config.h"
+#include "files.h"
+#include "nica/files.h"
+#include "systemd-class.h"
+
+/*
+ * This file implements 2-stage bootloader configuration in which shim is used as
+ * the first stage bootloader and systemd-boot as the second stage bootloader.
+ *
+ * This implementation uses the following ESP layout:
+ *
+ *     /EFI/
+ *          Boot/
+ *              BOOTX64.EFI         <-- this implementation never modifies the
+ *                                      default fallback loader
+ *
+ *      /org.clearlinux/
+ *          bootloaderx64.efi       <-- shim
+ *          loaderx64.efi           <-- systemd-boot bootloader
+ *          mmx64.efi               <-- MOK mgr
+ *          fbx64.efi               <-- MOK mgr
+ *
+ *          kernel/                 <-- kernels
+ *              kernel-org.clearlinux.....
+ *              ...
+ *
+ *      /loader/                    <-- systemd-boot config
+ *          entries/                <-- boot menu entries
+ *              Clear-linux...conf
+ *              ...
+ *          loader.conf             <-- bootloader config
+ *
+ * Note that default bootloader at /EFI/Boot/BOOTX64.EFI is never modified.
+ * Instead, we create an EFI BootXXX variable and put it first in the BootOrder
+ * EFI variable.
+ */
+
+static char *shim_systemd_get_kernel_dst(const BootManager *manager);
+static bool shim_systemd_install_kernel(const BootManager *manager, const Kernel *kernel);
+static bool shim_systemd_remove_kernel(const BootManager *manager, const Kernel *kernel);
+static bool shim_systemd_set_default_kernel(const BootManager *manager, const Kernel *kernel);
+static bool shim_systemd_needs_install(const BootManager *manager);
+static bool shim_systemd_needs_update(const BootManager *manager);
+static bool shim_systemd_install(const BootManager *manager);
+static bool shim_systemd_update(const BootManager *manager);
+static bool shim_systemd_remove(const BootManager *manager);
+static bool shim_systemd_init(const BootManager *manager);
+static void shim_systemd_destroy(const BootManager *manager);
+static int  shim_systemd_get_capabilities(const BootManager *manager);
+
+__cbm_export__ const BootLoader shim_systemd_bootloader = {
+        .name = "systemd",
+        .init = shim_systemd_init,
+        .get_kernel_dst = shim_systemd_get_kernel_dst,
+        .install_kernel = shim_systemd_install_kernel,
+        .remove_kernel = shim_systemd_remove_kernel,
+        .set_default_kernel = shim_systemd_set_default_kernel,
+        .needs_install = shim_systemd_needs_install,
+        .needs_update = shim_systemd_needs_update,
+        .install = shim_systemd_install,
+        .update = shim_systemd_update,
+        .remove = shim_systemd_remove,
+        .destroy = shim_systemd_destroy,
+        .get_capabilities = shim_systemd_get_capabilities
+};
+
+#if UINTPTR_MAX == 0xffffffffffffffff
```

Comment:
maybe this should be something that `configure` picks up? seems critical to know this up front.

Reply:
Yeah, there's like 25 places this is done. 26th with this one. `config.h` should really have them. I'm not sure this PR should be the place to fix this though.